account-kit/core/src/actions/getAccount.ts

```diff
@@ -28,17 +31,31 @@ export type GetAccountParams<TAccount extends SupportedAccountTypes> =
  * @returns {GetAccountResult<TAccount>} The result which includes the account if found and its status
  */
 export const getAccount = <TAccount extends SupportedAccountTypes>(
-  { type }: GetAccountParams<TAccount>,
+  { type, accountParams }: GetAccountParams<TAccount>,
   config: AlchemyAccountsConfig
 ): GetAccountResult<TAccount> => {
   const accounts = config.store.getState().accounts;
   const chain = getChain(config);
   const account = accounts?.[chain.id]?.[type];
   if (!account) {
-    return {
-      account: undefined,
-      status: "DISCONNECTED",
-    };
+    return defaultAccountState();
+  }
+
+  if (type === "ModularAccountV2" && account?.status === "READY") {
+    const accountConfig =
+      config.store.getState().accountConfigs[chain.id]?.[type];
+    const _accountConfig = accountConfig as
```

Comment:
❓ Why are we needing to cast the type here? What are the things tried with the trade offs?

I know that the state store for this section
<img width="556" alt="image" src="https://github.com/user-attachments/assets/d6df2168-693c-4b7a-a20d-5643ebce5281" />
Ends up being the very generic section, so I bring up the next thought. 
Could we use a guard to check and do this check/casting without the `as` for the account config?
Also, what's the type of mode. Maybe we could do all this checking and casting into a function for the mode?

Reply:
We could use the isMav2Config in the previous file?

Reply:
using a type guard seems like the best solution 👍